### PR TITLE
Task/add wechat unionpay specs

### PR DESCRIPTION
**Pull request by daniel-munch-cko**
Sits on top of bancontact to preserve order and to avoid merge conflicts. Once bancontact got merged this can be rebased.

### CODE REVIEW

File: spec/components/schemas/Payments/RequestSources/10_PaymentRequestKlarnaSource.yaml [hunk 1]
```diff
@@ -47,7 +45,7 @@ allOf:
         x-klarna-name: order_tax_amount
         x-klarna-docs: 'https://developers.klarna.com/api/#payments-api__create-a-new-order__order_tax_amount'
       products:
-        type: array
+        type: object
```
Comment (philippe-leonhardt-cko) [inline spec/components/schemas/Payments/RequestSources/10_PaymentRequestKlarnaSource.yaml]: should be array: https://developers.klarna.com/api/#payments-api-create-a-new-order

File: spec/components/schemas/Payments/PaymentRequestSource.yaml [hunk 2]
```diff
@@ -18,6 +18,9 @@ discriminator:
     safetypay: '#/components/schemas/13_PaymentRequestSafetyPaySource'
     sofort: '#/components/schemas/14_PaymentRequestSofortSource'
     bancontact: '#/components/schemas/15_PaymentRequestBancontactSource'
+    wechat-qr: '#/components/schemas/16_PaymentRequestWeChat QRSource'
+    wechat-web: '#/components/schemas/17_PaymentRequestWeChat WebSource'
+    unionpay: '#/components/schemas/18_PaymentRequestUnionpaySource'
```
Comment (kevin-reece-cko) [inline spec/components/schemas/Payments/PaymentRequestSource.yaml]: should be unionpay before wechat (i.e. alphabetical). See here: https://github.com/checkout/checkout-api-reference/pull/43/files#r271377820

File: spec/components/schemas/Payments/PaymentResponseSource.yaml [hunk 3]
```diff
@@ -15,6 +15,9 @@ discriminator:
     safetypay: '#/components/schemas/09_PaymentResponseSafetyPaySource'
     sofort: '#/components/schemas/10_PaymentResponseSofortSource'
     bancontact: '#/components/schemas/15_PaymentResponseBancontactSource'
+    wechat-qr: '#/components/schemas/16_PaymentResponseWeChat QRSource'
+    wechat-web: '#/components/schemas/17_PaymentResponseWeChat WebSource'
+    unionpay: '#/components/schemas/18_PaymentResponseUnionpaySource'
```
Comment (kevin-reece-cko) [inline spec/components/schemas/Payments/PaymentResponseSource.yaml]: should be unionpay before wechat (i.e. alphabetical). See here: https://github.com/checkout/checkout-api-reference/pull/43/files#r271377820

File: spec/components/schemas/Payments/RequestSources/10_PaymentRequestKlarnaSource.yaml (hunk 1 above)
Comment (kevin-reece-cko) [reply spec/components/schemas/Payments/RequestSources/10_PaymentRequestKlarnaSource.yaml]: This is my fault. The Klarna changes to fix this need to be merged (as in internal Klarna PR 180: https://github.com/CKOTech/checkout-ap-klarna/pull/180).

**Pull request closed without merging**